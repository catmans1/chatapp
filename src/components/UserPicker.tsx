import React, {useState} from 'react';
import {StyleSheet} from 'react-native';
import styled from 'styled-components/native';
import Icon from 'react-native-vector-icons/AntDesign';

import ModalDropdown from '../libs/DropDown/ModalDropdown';
import useUser from '../hooks/useUser';
import {UserList} from '../fakers';
import {useEffect} from 'react';

export default function UserPicker() {
  const {user, setUser} = useUser();

  const [value, setValue] = useState<string>('');
  const [open, setOpen] = useState<boolean>(false);

  useEffect(() => {
    const currentUser = user?.length > 0 ? user : UserList[0];
    setValue(currentUser);
  }, [user]);

  function onHandleSelect(_: string, option: string) {
    if (value !== option) {
      setValue(option);
      setUser(option);
    }
  }

  return (
    <ModalDropdown
      defaultValue={value}
      onSelect={onHandleSelect}
      options={UserList}
      style={styles.container}
      textStyle={styles.dropDownText}
      dropdownStyle={styles.dropDownContainer}
      renderRightComponent={() => {
        return <Icon name={open ? 'up' : 'down'} size={14} />;
      }}
      renderRow={(option: string) => {
        return <TextPicker>{option}</TextPicker>;
      }}
      onDropdownWillShow={() => setOpen(true)}
      onDropdownWillHide={() => setOpen(false)}
    />
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: 40,
    marginHorizontal: 10,
    marginVertical: 10,
    paddingHorizontal: 10,
    paddingVertical: 10,
    borderRadius: 6,
    borderColor: '#c4c0c0',
    borderWidth: 1,
    justifyContent: 'center',
  },
  dropDownContainer: {
    height: UserList.length * 40,
    width: '95%',
    marginTop: 15,
    marginLeft: -10,
    borderRadius: 6,
    borderColor: '#c4c0c0',
    borderWidth: 1,
    justifyContent: 'center',
  },
  dropDownText: {
    flex: 1,
    color: 'black',
    fontSize: 16,
  },
});

const TextPicker = styled.Text`
  color: black;
  padding-top: 10px;
  padding-bottom: 10px;
  padding-left: 10px;
  font-size: 16px;
`;
