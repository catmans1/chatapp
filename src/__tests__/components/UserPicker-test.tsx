import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';

import UserPicker from '../../components/UserPicker';

it('renders correctly', () => {
  const tree = renderer.create(<UserPicker />).toJSON();
  expect(tree).toMatchSnapshot();
});
