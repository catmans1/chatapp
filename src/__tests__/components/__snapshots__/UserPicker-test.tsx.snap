// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  animated={true}
  defaultIndex={-1}
  defaultValue=""
  disabled={false}
  dropdownStyle={
    Object {
      "borderColor": "#c4c0c0",
      "borderRadius": 6,
      "borderWidth": 1,
      "height": 120,
      "justifyContent": "center",
      "marginLeft": -10,
      "marginTop": 15,
      "width": "95%",
    }
  }
  isFullWidth={false}
  keySearchObject="label"
  keyboardShouldPersistTaps="never"
  multipleSelect={false}
  numberOfLines={1}
  onDropdownWillHide={[Function]}
  onDropdownWillShow={[Function]}
  onSelect={[Function]}
  options={
    Array [
      "Joyse",
      "Russell",
      "Sam",
    ]
  }
  renderButtonComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  renderRightComponent={[Function]}
  renderRow={[Function]}
  renderRowComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  saveScrollPosition={true}
  scrollEnabled={true}
  searchPlaceholder="Search"
  showSearch={false}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "borderColor": "#c4c0c0",
      "borderRadius": 6,
      "borderWidth": 1,
      "flex": 1,
      "height": 40,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "marginVertical": 10,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
  textStyle={
    Object {
      "color": "black",
      "flex": 1,
      "fontSize": 16,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "black",
                "flex": 1,
                "fontSize": 16,
              },
              undefined,
            ],
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Óòù
      </Text>
    </View>
  </View>
</View>
`;
